identity/backend/src/main/java/io/camunda/identity/usermanagement/service/GroupService.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.identity.usermanagement.service;
+
+import io.camunda.identity.usermanagement.Group;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
+
+@Service
+public class GroupService {
+  private final CamundaUserDetailsManager userDetailsManager;
+
+  public GroupService(final CamundaUserDetailsManager userDetailsManager) {
+    this.userDetailsManager = userDetailsManager;
+  }
+
+  public List<Group> findAllGroups() {
+    return userDetailsManager.findAllGroups().stream()
+        .map(g -> new Group(userDetailsManager.findGroupId(g), g))
```

Comment:
A valid point, I did it because I wanted to reuse the existing methods but as It's a findAll method it may downgrade performance a lot, I try to change it as you suggested.